pkg/store/memstore/memstore.go

```diff
@@ -2,17 +2,23 @@ package memstore
 
 import (
 	"context"
+	"math"
 	"sync"
+	"time"
 
+	"github.com/openshift/telemeter/pkg/store"
 	clientmodel "github.com/prometheus/client_model/go"
 )
 
+const ttl = 20 * time.Minute
```

Comment:
One this strikes me as funny, in the telemeter server, we hardcode the maximum sample age of a federated metric to be:
```go
// maxSampleAge is the maximum age of a sample that we can report via federation.
const maxSampleAge = 10 * time.Minute
```

This means, we never report any metrics older than 10 minutes, but with this TTL we will still hold on to metrics as old as 20 minutes. Why? Why do we want the ability to hold on to metrics that we will never report? Furthermore, the semantics of the opposite, that is, having a TTL that is shorter than the max age are likewise odd, e.g. we could have a TTL of 5 minutes and a max age of 10 minutes.

Why not use the max age as the TTL value for cleanups?

Reply:
I was actually thinking the same. The only reason to keep it here was to have some additional margin for the data to live in the store in the case the consumer has to retry moving `maxSampleAge` further out in time. I am actually fine with merging it, wdyt?

Reply:
Even if the consumer had to retry fetching metrics, if the time of the metric was greater than the max age, then it doesn't matter how long the TTL is, the metric will not be returned. If the metric timestamp is less than the max age then the metrics will be returned. So in this case there is still no benefit to having a TTL that is longer than the max age.

I think we should merge this for now to simplify; there are a lot of weird edge cases and unspecified behaviors currently depending on the relative values of these constants.

Reply:
agreed, merged :+1: